sdk/python/tests/unit/api/test_search_api.py

```diff
@@ -0,0 +1,2407 @@
+            "/search?query=user&projects=nonexistent1&projects=nonexistent2"
+        )
+        assert response.status_code == 200
+
+        data = response.json()
+        assert data["projects_searched"] == []  # no existing projects to search
+        # Should return empty results since projects don't exist
+        assert data["results"] == []
+        assert not data["pagination"].get("totalCount", False)
+        assert (
+            data["error"]
+            == "Following projects do not exist: nonexistent1, nonexistent2"
+        )
+
+    def test_search_mixed_existing_nonexistent_projects(self, search_test_app):
+        """Test searching in mix of existing and non-existing projects"""
+        response = search_test_app.get(
+            "/search?query=user&projects=test_project&projects=nonexistent_project"
+        )
+        assert response.status_code == 200
+
+        data = response.json()
+        assert data["projects_searched"] == ["test_project"]  # only existing project
+        assert data["error"] == "Following projects do not exist: nonexistent_project"
+
+        # Should only find results from existing project
+        for result in data["results"]:
+            if "project" in result:
+                assert result["project"] == "test_project"
+
+    def test_search_many_projects_performance(self, search_test_app):
+        """Test search performance with many projects"""
+        # Create a list of many projects (mix of existing and non-existing)
+        fake_projects = [f"fake_project_{i}" for i in range(20)]
+        many_projects = ["test_project"] + fake_projects
+        projects_param = "&".join([f"projects={p}" for p in many_projects])
+
+        response = search_test_app.get(f"/search?query=user&{projects_param}")
+        assert response.status_code == 200
+
+        data = response.json()
+        assert len(data["projects_searched"]) == 1  # only 1 real project exists
+        assert "test_project" in data["projects_searched"]
+        assert data["error"] == "Following projects do not exist: " + ", ".join(
+            fake_projects
+        )
+
+        # Should still return results from the one existing project
+        if data["results"]:
+            for result in data["results"]:
+                if "project" in result:
+                    assert result["project"] == "test_project"
+
+    def test_search_duplicate_projects_deduplication(self, search_test_app):
+        """Test that duplicate projects in list are handled properly"""
+        response = search_test_app.get(
+            "/search?query=user&projects=test_project&projects=test_project&projects=test_project"
+        )
+        assert response.status_code == 200
+
+        data = response.json()
+        # API should handle duplicates gracefully (may or may not deduplicate)
+        # At minimum, should not crash and should search test_project
+        assert "test_project" in data["projects_searched"]
```

Comment:
I thnk we should check if the data is not repeating for same project and/or length of response it 1.